From b9fd721c02a4aa603cd508b6f6390cc58652ebeb Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Tue, 3 Mar 2015 16:24:30 +0100
Subject: [PATCH] manager: try to create virtual devices on device state changes

vlans can only be created on bonds that have slaves -- it's not sufficient to
attempt their creation when devices are added.
---
 src/nm-manager.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/nm-manager.c b/src/nm-manager.c
index 1970c58..c6d4562 100644
--- a/src/nm-manager.c
+++ b/src/nm-manager.c
@@ -540,6 +540,8 @@ update_active_connection_timestamp (NMManager *manager, NMDevice *device)
 	g_key_file_free (timestamps_file);
 }
 
+static void system_create_virtual_devices (NMManager *self);
+
 static void
 manager_device_state_changed (NMDevice *device,
                               NMDeviceState new_state,
@@ -565,6 +567,8 @@ manager_device_state_changed (NMDevice *device,
 
 	if (new_state == NM_DEVICE_STATE_ACTIVATED)
 		update_active_connection_timestamp (manager, device);
+
+	system_create_virtual_devices (manager);
 }
 
 /* Removes a device from a device list; returns the start of the new device list */
-- 
1.7.1

